### PR TITLE
[Calling] Applying capability for Reaction feature

### PR DESCRIPTION
**Pull request by Mohtasim**
# What

* Reactions will not render on video tiles if meeting organizer policy is set to false.
* Reaction button will be hidden from UI

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md
Comment (Mohtasim) [inline packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:327]: Looking for a better suggestion for this property name. 
This selector property reflects the meeting organizer policy on sending/receiving Reaction Capability in a teams interop meeting.

Comment (JamesBurnside) [inline packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:327]: As this is just used for the composite, remove from the component bindings. The reason for this is that it is *very* app-specific behavior. So if someone if using raw components we don't want to dictate this behavior. As a rule of thumb, anything in the selector should map directly to a prop on the react-components' interface

File: packages/calling-stateful-client/src/CallContext.ts
Comment (JamesBurnside) [inline packages/calling-stateful-client/src/CallContext.ts:1100]: Should this be?
```suggestion
      capabilitiesFeature.capabilities.useReactions.reason !== 'FeatureNotSupported')
```

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md
Comment (JamesBurnside) [reply packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:327]: As this is just used for the composite, remove from the component bindings. As a rule of thumb, anything in the selector should map directly to a prop on the react-components' interface

Comment (JamesBurnside) [reply packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:327]: Instead you can create a new selector under callcomposite/selectors that can retrieve this property

File: packages/calling-stateful-client/src/CallContext.ts
Comment (JamesBurnside) [reply packages/calling-stateful-client/src/CallContext.ts:1100]: Oh wait, is this saying "if the feature capability is not supported", i.e. we don't know the status of the capability for this feature?

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md
Comment (Mohtasim) [reply packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:327]: Got it. Thanks :)

File: packages/calling-stateful-client/src/CallContext.ts
Comment (Mohtasim) [reply packages/calling-stateful-client/src/CallContext.ts]: Yes. For that `FeatureNotSupported` case, it would be on by default. This would be the case for ACS group & Rooms calls.

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md
Comment (Mohtasim) [reply packages/calling-component-bindings/review/beta/calling-component-bindings.api.md]: So I've refactored and removed the `allowed` property and created a new selector in react-composite package to decide the hide/show of the button in the control bar according to capability.

File: packages/calling-stateful-client/src/CallContext.ts
Comment (JamesBurnside) [inline packages/calling-stateful-client/src/CallContext.ts:420]: If I'm not capable to react, I think we should still be able to see others' reactions?

Comment (JamesBurnside) [reply packages/calling-stateful-client/src/CallContext.ts:420]: (e.g. if I can't unmute or share video, I can still hear/see others)

Comment (Mohtasim) [reply packages/calling-stateful-client/src/CallContext.ts:420]: When Capability is turned off in a meeting, None can react. Hence local participant will not receive any reactions as well. 
Since, Web SDK establishes a socket connection through the live state service it can still propagate reaction messages since the Live State Service is unaware of the capability changes or updates. 

So in the teams interop scenario, users joining with UI library or their own acs app which uses SDK APIs, with wrong implementation can pass reaction message. So clients using teams app wont see the reactions but CTE's using UI lib will be able to see the reaction which might become a privacy concern. To have parity, implying this check. 

Web SDK does handle capability changes. But this can happen in Teams Group Calls since Web SDK's teams group call scenarios & 1 to 1 call handling with capability is not stable yet.

Comment (JamesBurnside) [reply packages/calling-stateful-client/src/CallContext.ts:420]: > with wrong implementation can pass reaction message .... CTE's using UI lib will be able to see the reaction which might become a privacy concern.

Developers won't have this level of knowledge when building their own apps direct from the SDK to know they should be ignoring inbound reactions if the capability is set. If this is in any way a privacy issue we must handle this at the SDK level. What's the limitation of the SDK performing the same check here on the capability and not sending the reaction?

Comment (Mohtasim) [reply packages/calling-stateful-client/src/CallContext.ts:420]: Yes I agree. The feature in SDK checks for Capability while send or receiving reaction. 
The capability API does not handle cases for 1 to 1 & 1 to N teams group calls. We have handled these cases without capability API and made it stable. 
With more changing of Capability feature and API and removing direct tscall event listening from the feature, will make it problematic.

**Review by dmceachernmsft — APPROVED** [review]

**Review by JamesBurnside — APPROVED** [review]
Probably update the default capability in the mock adapter to be true instead of false to reduce the snapshot changes. Or maybe keep just attendee as false? Either way PR looks good so approving, no need for re-approval if you just change that capability value or not

**Pull request merged**
